### PR TITLE
Fix Range#size results

### PR DESCRIPTION
**Pull request by ksss**
# Actual

```rb
$ mirb
mirb - Embeddable Interactive Ruby Shell

> (1.0..15.9).size
 => 16
> (0..Float::INFINITY).size
 => -2147483648
```

# Expect

```rb
$ irb
irb(main):001:0> (1.0..15.9).size
=> 15
irb(main):002:0> (0..Float::INFINITY).size
=> Infinity
```

This fix passed all specs of that https://github.com/ruby/spec/blob/66daf290cb49707190481b4cf24e700010cd53bf/core/range/size_spec.rb

### CODE REVIEW

File: mrbgems/mruby-range-ext/src/range.c [hunk 1]
```diff
@@ -162,14 +164,27 @@ mrb_range_size(mrb_state *mrb, mrb_value range)
...
-    double f;
-
-    if (beg_f > end_f) return mrb_fixnum_value(0);
-    f = end_f - beg_f;
-    if (!r->excl) {
-      return mrb_fixnum_value((mrb_int)ceil(f + 1));
+#ifndef DBL_EPSILON
+#define DBL_EPSILON 2.2204460492503131e-16
+#endif
+    double n = end_f - beg_f;
```
Comment (dabroz) [inline mrbgems/mruby-range-ext/src/range.c]: Why not `mrb_float`?

File: mrbgems/mruby-range-ext/src/range.c [hunk 2]
```diff
@@ -162,14 +164,27 @@ mrb_range_size(mrb_state *mrb, mrb_value range)
     num_p = FALSE;
   }
   if (num_p) {
-    double f;
-
-    if (beg_f > end_f) return mrb_fixnum_value(0);
-    f = end_f - beg_f;
-    if (!r->excl) {
-      return mrb_fixnum_value((mrb_int)ceil(f + 1));
+#ifndef DBL_EPSILON
```
Comment (dabroz) [inline mrbgems/mruby-range-ext/src/range.c]: We already include <float.h> (ANSI C), which includes this constant.

File: mrbgems/mruby-range-ext/src/range.c (hunk 1 above)
Comment (ksss) [inline mrbgems/mruby-range-ext/src/range.c]: I did'n know that.
Should we use `mrb_float` instead of `double` and `float` in all cases?

**Comment by matz** [comment]
mrb_float would be better, although I saw a lot of doubles in old code.

**Comment by ksss** [comment]
Thank you, I fixed it!

**Pull request merged**